core/src/labels/labelCollider.cpp

```diff
@@ -190,7 +204,7 @@ void LabelCollider::process(TileID _tileID, float _tileInverseScale, float _tile
             repeatGroup = l1->options().repeatGroup;
 
             if (repeatGroup != 0) {
-                handleRepeatGroup(pair.first);
+                repeatPos = filterRepeatGroups(repeatPos, pair.first);
```

Comment:
`repeatPos` is a little ambiguous (it seems to be where we last stopped? `lastFilteredIndex` or something on those lines seems more appropriate).